service/service.go

```diff
@@ -113,7 +114,13 @@ func New(ctx context.Context, set Settings, cfg Config) (*Service, error) {
 		MetricsLevel:   cfg.Telemetry.Metrics.Level,
 		// Construct telemetry attributes from build info and config's resource attributes.
 		Resource: pcommonRes,
-		Status:   status.NewReporter(srv.host.notifyComponentStatusChange),
+		Status: status.NewReporter(srv.host.notifyComponentStatusChange, func(err error) {
+			if errors.Is(err, status.ErrStatusNotReady) {
+				srv.telemetry.Logger().Warn("Invalid transition", zap.Error(err))
+			} else {
+				srv.telemetry.Logger().Debug("Invalid transition", zap.Error(err))
```

Comment:
I'm wondering if we might omit logging about invalid transitions altogether since they are expected to happen and do not indicate an underlying problem using the API.

Reply:
sure. I can do that.

Now the bigger problem is how to deprecate this method to remove the error returned - I'll work on that next.